src/main/java/org/folio/rest/impl/InventoryHelper.java

```diff
@@ -173,11 +173,17 @@
     itemRecord.put(ITEM_STATUS, new JsonObject().put(ITEM_STATUS_NAME, receivedItem.getItemStatus()));
     if (StringUtils.isNotEmpty(receivedItem.getBarcode())) {
       itemRecord.put(ITEM_BARCODE, receivedItem.getBarcode());
+    } else if (isOnOrderItemStatus(receivedItem)) {
+      itemRecord.remove(ITEM_BARCODE);
```

Comment:
We don't want to remove the barcode, just change the status

Reply:
Corrected